admin/app/controllers/workarea/admin/create_release_undos_controller.rb

```diff
@@ -13,10 +13,15 @@ def create
         @undo_release.attributes = params[:release]
 
         if @undo_release.save
-          @release.changesets.each do |changeset|
+          @release.changesets.limit(Workarea.config.per_page).each do |changeset|
             changeset.build_undo(release: @undo_release.model).save!
           end
 
+          BuildReleaseUndoChangesets.perform_async(
+            @undo_release.id,
+            @release.id
+          ) if @release.show_changeset_summary?
```

Comment:
I see the logic reuse, but this method is tied to display so if it were to get decorated or something, changing this logic would definitely be a surprise.

```suggestion
          ) if @release.changeset_count > Workarea.config.per_page
```